### ISSUE TITLE
Email::Simple::Creator constructor doesn't folder headers

### ISSUE THREAD

**Issue by aap-matt** [rjbs/Email-Simple#7]
Hello,

Yesterday I ran afoul of an unexpected issue with Email::Simple::Creator (v2.203) that I wanted to make you aware of. I think I'd consider this a bug, though perhaps I'm missing a piece of documentation somewhere that would explain the behavior I'm seeing.

When I set a header via Email::Simple->create the header does not get folded. However, if I use header_set the header does get folded. It seems like headers ought to get folded when called via the constructor, but I can easily work around the issue by calling header_set to achieve the folding I require.

Here's an example script illustrating the issue:

```perl
use Email::Simple;
use strict;

my $to   = 'to@example.com';
my $from = 'from@example.com';

my $subject = 'A ' x 50; # Long enough to need to be folded

my $email_1 = Email::Simple->create(
  header => [
    To      => $to,
    From    => $from,
    Subject => $subject, # string specified in constructor does *not* get folded
  ]
);

print "EMAIL 1:\n".$email_1->as_string;


my $email_2 = Email::Simple->create(
  header => [
    To      => $to,
    From    => $from,
  ]
);
$email_2->header_set('Subject' => $subject); # using header_set *does* cause string to get folded

print "EMAIL 2:\n".$email_2->as_string;
```

**Comment by rjbs** [comment]
I think this is an unforeseen consequence of some refactoring that was done a while ago to avoid re-wrapping headers needlessly, plus the goofy way that new emails are built.

While I'd like to fix this, eventually, is it causing a real problem?  Header folding, especially in non-MIME messages, should be entirely optional.

**Comment by aap-matt** [comment]
Thanks for the response.

This caused an issue for me when I was using Email::Simple to create an SMTP message for sending to a Transactional Email service. Their SMTP Headers API doc had some warnings about what would happen if a particular header line was too long, so I was wanting to be very sure my long header lines were getting folded. I was simply confused that the headers I specified in the constructor were not folded, while ones that were added via header_set were.

This was easy enough for me to work around -- I just ensure any headers that will be long enough to need folding are aded via header_set. So, I would say this issue isn't causing a "real" problem for me since I can easily work around it. However, because of the inconsistency, it does seem like a bug -- but I'd be fine with considering this a low priority issue, or perhaps just a caveat mentioned in the docs somewhere.
